Xamarin.Forms.Platform.Android/Renderers/DatePickerRenderer.cs

```diff
@@ -123,9 +127,16 @@ void OnTextFieldClicked()
 
 			UpdateMinimumDate();
 			UpdateMaximumDate();
+
+			_dialog.CancelEvent += OnCancelButtonClicked;
 			_dialog.Show();
 		}
 
+		void OnCancelButtonClicked(object sender, EventArgs e)
+		{
+			Element.Unfocus();
```

Comment:
ClearFocus is on the Android View class and calling Element.Unfocus is keeping track of the fact that it's being dismissed/unfocused away since it apparently wasn't being updated by the cancel button which I suppose is treated differently than clicking Ok/Done or the equivalent.